src/slic3r/GUI/AMSMaterialsSetting.cpp

```diff
@@ -992,10 +992,11 @@ void AMSMaterialsSetting::on_select_filament(wxCommandEvent &evt)
                 auto filament_item = map_filament_items[m_comboBox_filament->GetValue().ToStdString()];
                 std::string filament_id   = filament_item.filament_id;
                 if (it->filament_id.compare(filament_id) == 0) {
+                    ConfigOption *       printer_opt  = it->config.option("compatible_printers");
+                    ConfigOptionStrings *printer_strs = dynamic_cast<ConfigOptionStrings *>(printer_opt);
                     bool has_compatible_printer = false;
-                    std::string preset_name            = it->name;
-                    for (std::string printer_name : printer_names) {
-                        if (preset_name.find(printer_name) != std::string::npos) {
+                    for (auto printer_str : printer_strs->values) {
```

Comment:
Should we take care of the case where the compatible printers are empty, which is the case if the filament is from the Orca Filament Library?

Reply:
It might actually be tricky to handle that case. I think it's fine for now. We don't need to do it. At the moment, all global profiles have overridden version for BBL anyway.